src/security/FactoryGuardian.sol

```diff
@@ -0,0 +1,112 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+pragma solidity ^0.8.0;
+
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "./BaseGuardian.sol";
+
+/**
+ * @title Factory Guardian
+ * @dev This module provides a mechanism that allows authorized accounts to trigger an emergency stop
+ *
+ */
+abstract contract FactoryGuardian is BaseGuardian {
+    /*
+    //////////////////////////////////////////////////////////////
+                            EVENTS
+    //////////////////////////////////////////////////////////////
+    */
+
+    event PauseUpdate(address account, bool createPauseUpdate, bool liquidatePauseUpdate);
+
+    /*
+    //////////////////////////////////////////////////////////////
+                            STORAGE
+    //////////////////////////////////////////////////////////////
+    */
+    bool public createPaused;
+    bool public liquidatePaused;
+
+    constructor() {}
+
+    /*
+    //////////////////////////////////////////////////////////////
+                            MODIFIERS
+    //////////////////////////////////////////////////////////////
+    */
+
+    /**
+     * @dev This modifier is used to restrict access to certain functions when the contract is paused for create vault.
+     * It throws if create vault is paused.
+     */
+    modifier whenCreateNotPaused() {
+        require(!createPaused, "Guardian: create paused");
+        _;
+    }
+
+    /**
+     * @dev This modifier is used to restrict access to certain functions when the contract is paused for liquidate vaultq.
+     * It throws if liquidate vault is paused.
+     */
+    modifier whenLiquidateNotPaused() {
+        require(!liquidatePaused, "Guardian: liquidate paused");
+        _;
+    }
+
+    /**
+     * @notice This function is used to pause the contract.
```

Comment:
this could be an @inheritdoc BaseGuardian, although we don't do this consistently anyway